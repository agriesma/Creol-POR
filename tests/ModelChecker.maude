***
*** Test suite for the model checker.
***
set show timing off .
set show advise off .
load modelchecker .

quit
