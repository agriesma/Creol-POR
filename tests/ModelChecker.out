==========================================
reduce in TEST-TAILCALL : init .
result Configuration: 
< "Tailcall" : Cl | Inh: noInh, Par: "x", Att: "x" |-> null, Mtds:
  < "init" : Mtdname | Param: noVid, Latt: noSubst, Code: return(emp) > *
  < "run" : Mtdname | Param: noVid, Latt: noSubst, Code: if "<"(int(0) :: "x")
    th "x" ::= "-"("x" :: int(1)) ; tailcall "run"(emp) el return(emp) fi >,
  Ocnt: 1 >
 
< ob("Tailcall0") : "Tailcall" |
  Att: "this" |-> ob("Tailcall0"), "x" |-> int(5),
  Pr: (".class" |-> str("Tailcall"), ".label" |-> label(ob("Tailcall0"), ob(
    "Tailcall0"), "init", emp), ".method" |-> str("init"), "caller" |-> ob(
    "Tailcall0")), cont(label(ob("Tailcall0"), ob("Tailcall0"), "init", emp)),
  PrQ: ".anon" |-> label(ob("Tailcall0"), ob("Tailcall0"), "init", emp), label(
    ob("Tailcall0"), ob("Tailcall0"), "init", emp) ?(noVid) ; ".anon" ! "run"(
    emp) ; free(".anon"),
  Dealloc: noDealloc,
  Ev: comp(label(ob("Tailcall0"), ob("Tailcall0"), "init", emp), emp),
  Lcnt: 0 >
 
< ob("main") : "" |
  Att: noSubst,
  Pr: idle,
  PrQ: noProc,
  Dealloc: noDealloc,
  Ev: noMsg,
  Lcnt: 0 >

==========================================
rewrite [1] in TEST-TAILCALL : init .
result Configuration: 
< "Tailcall" : Cl | Inh: noInh, Par: "x", Att: "x" |-> null, Mtds:
  < "init" : Mtdname | Param: noVid, Latt: noSubst, Code: return(emp) > *
  < "run" : Mtdname | Param: noVid, Latt: noSubst, Code: if "<"(int(0) :: "x")
    th "x" ::= "-"("x" :: int(1)) ; tailcall "run"(emp) el return(emp) fi >,
  Ocnt: 1 >
 
< ob("Tailcall0") : "Tailcall" |
  Att: "this" |-> ob("Tailcall0"), "x" |-> int(5),
  Pr: ".anon" |-> label(ob("Tailcall0"), ob("Tailcall0"), "init", emp), label(
    ob("Tailcall0"), ob("Tailcall0"), "init", emp) ?(noVid) ; ".anon" ! "run"(
    emp) ; free(".anon"),
  PrQ: noProc,
  Dealloc: noDealloc,
  Ev: comp(label(ob("Tailcall0"), ob("Tailcall0"), "init", emp), emp),
  Lcnt: 0 >
 
< ob("main") : "" |
  Att: noSubst,
  Pr: idle,
  PrQ: noProc,
  Dealloc: noDealloc,
  Ev: noMsg,
  Lcnt: 0 >

==========================================
rewrite in TEST-TAILCALL : init .
result Configuration: 
< "Tailcall" : Cl | Inh: noInh, Par: "x", Att: "x" |-> null, Mtds:
  < "init" : Mtdname | Param: noVid, Latt: noSubst, Code: return(emp) > *
  < "run" : Mtdname | Param: noVid, Latt: noSubst, Code: if "<"(int(0) :: "x")
    th "x" ::= "-"("x" :: int(1)) ; tailcall "run"(emp) el return(emp) fi >,
  Ocnt: 1 >
 
< ob("Tailcall0") : "Tailcall" |
  Att: "this" |-> ob("Tailcall0"), "x" |-> int(0),
  Pr: idle,
  PrQ: noProc,
  Dealloc: noDealloc,
  Ev: noMsg,
  Lcnt: 0 >
 
< ob("main") : "" |
  Att: noSubst,
  Pr: idle,
  PrQ: noProc,
  Dealloc: noDealloc,
  Ev: noMsg,
  Lcnt: 0 >

Bye.
